### PR TITLE
[docs] Update "What's new" page

### PR DESCRIPTION
**Pull request by cherniavskii**
Preview: https://deploy-preview-14858--material-ui-x.netlify.app/x/whats-new/

### CODE REVIEW

File: docs/src/modules/components/WhatsNewLayout.js [hunk 1]
```diff
@@ -17,6 +17,94 @@ import TimelineDot from '@mui/lab/TimelineDot';
...
+        url: '/x/react-data-grid/tree-data/#automatic-parents-and-children-selection',
+      },
+      {
+        title: 'Date and Time Pickers – Support date-fns v4',
+        url: '/x/react-date-pickers/adapters-locale/#with-date-fns',
+      },
+    ],
+  },
+  {
+    title: 'MUI X v7.15',
```
Comment (cherniavskii) [inline docs/src/modules/components/WhatsNewLayout.js:41]: I decided to put 7.15 separately to highlight Material UI v6 support.

File: docs/src/modules/components/WhatsNewLayout.js
Comment (cherniavskii) [inline docs/src/modules/components/WhatsNewLayout.js:1]: Feel free to suggest any highlights you'd like to include!

cc @joserodolfofreitas  @mui/xgrid @mui/explore @mui/xcharts

File: docs/src/modules/components/WhatsNewLayout.js [hunk 2]
```diff
@@ -17,6 +17,94 @@ import TimelineDot from '@mui/lab/TimelineDot';
...
+    date: new Date(2024, 9, 4),
+    url: 'https://github.com/mui/mui-x/releases/tag/v7.19.0',
+    highlightList: [
+      {
+        title: 'Data Grid – Row spanning',
+        url: '/x/react-data-grid/row-spanning/',
+      },
+      {
+        title: 'Data Grid – Automatic parents and children selection',
+        url: '/x/react-data-grid/tree-data/#automatic-parents-and-children-selection',
```
Comment (MBilalShafi) [inline docs/src/modules/components/WhatsNewLayout.js:32]: MIght be better to directly link the main docs.
```suggestion
        url: '/x/react-data-grid/row-grouping/#automatic-parents-and-children-selection',
```

**Review by arminmeh — APPROVED** [review]

**Review by MBilalShafi — APPROVED** [review]

**Review by joserodolfofreitas — COMMENTED** [review]
Great initiative, @cherniavskii. Thank you!

**Pull request merged**